flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -646,6 +653,22 @@ static constexpr IntrinsicHandler handlers[]{
      /*isElemental=*/true},
 };
 
+static const IntrinsicHandler *findIntrinsicHandler(llvm::StringRef name) {
```

Comment:
Making it a binary search looks a great idea to me. Your algo looks correct to me, but I think from a maintenance/llvm style point of view, it might be best to use C++ tools to do this. I think you can do:
```
  auto compare = [](const IntrinsicHandler &handler, llvm::StringRef nm) {
    return nm.compare(handler.name) > 0;
  };
  auto match =
      std::lower_bound(std::begin(handlers), std::end(handlers), name, compare);
  if (match != std::end(handlers) && match->name == name)
    return match;
  return nullptr;
```

`std::lower_bound` does a binary search assuming ordering of the container.

Reply:
Ah, thanks, I didn't realize that std::begin/end were applicable to arrays.  I'll try this.